lib/Router.js

```diff
@@ -0,0 +1,198 @@
+
+	this.initialize = function () {
+		var promise = new Promise(function (resolve, reject) {
+			/*
+			*On startup, must populate the conversations object with all of the
+			*current conversations going on. Done by getting all numbers associated
+			*with this application and parsing the client and stranger numbers from the
+			*'name' field in each phoneNumber.
+			*/
+			catapult.PhoneNumber.list(client, {
+			applicationId : applicationId,
+			numberState   : "ENABLED" }, function (err, numbers) {
+				if (err) {
+					reject(err);
+				}
+				else {
+					self.myNumbers = numbers;
+					for (var i = 0; i < numbers.length; i += 1){
+						if (numbers[i].name){
+							conversations[numbers[i].number] = JSON.parse(numbers[i].name);
+						}
+						else {
+							conversations[numbers[i].number] = [ null, null, null ];
+						}
+					}
+					resolve();
+				};
+			});
+		});
+		return promise;
+	};
+
+	router.use(bodyParser.json());
+	router.use("/", express.static("public"));
+
+	/*
+	*iterates through the list of PhoneNumber objects to find the one with the matching
+	*'number' field.
+	*/
+	this.getNumber = function (number) {
+		for (var i = 0; i < this.myNumbers.length; i += 1){
+			if (this.myNumbers[i].number === number){
+				return this.myNumbers[i];
+			}
+		}
+		return null;
+	}
+
+	/*
+	*Catapult post message to search and return a list of numbers matching city and state
+	*/
+	this.searchNumbers = function (city, state) {
+		var promise = new Promise(function (resolve,reject) {
+			catapult.AvailableNumber.searchLocal(client, { state : state, city : city }, function (err, numbers) {
+				if (err) {
+					reject(err);
+				}
+				else {
+					resolve(numbers);
+				}
+			});
+		});
+		return promise;
+	}
+
+	/*
+	*Catapult call to buy a phone number and set the applicationId to this app
+	*/
+	this.orderNumber = function (number) {
+		var promise = new Promise(function (resolve, reject) {
+			catapult.PhoneNumber.create(client, { number : number, applicationId : applicationId }, function (err, phoneNumber) {
+				if (err) {
+					reject(err);
+				}
+				else {
+					resolve(phoneNumber);
+				}
+			});
+		});
+		return promise;
+	}
+
+	this.sendTextFromAlias = function (alias, number, message) {
+		var promise = new Promise(function (resolve, reject) {
+			catapult.Message.create(client, { from : alias, to : number, text : message }, function (err, message) {
+				if (err) {
+					reject(err);
+				}
+				else {
+					resolve();
+				}
+			});
+		});
+		return promise;
+	}
+
+	/*
+	*POSTs to /numbers come from the web interface and contain the fields
+	*'text', 'oneLiner', 'city', and 'state'. On receiving a POST to /numbers, the app searches
+	*for a new number in the city and state specified, orders it, and sends the client
+	*a text from the alias number with instructions.
+	*/
+	router.post("/numbers", function (req, res) {
+		var clientNum = req.body.text;
+		var city = req.body.city;
+		var state = req.body.state;
```

Comment:
What happens if you receive a POST that does not have these fields set (or if the user doesn't fill in those fields on your web form)? Does your search fail? Is this failure accurately relayed to the client as a bad request?